xds/src/main/java/io/grpc/xds/CdsLoadBalancer.java

```diff
@@ -181,6 +181,7 @@ public boolean canHandleEmptyAddressListFromNameResolution() {
         public void shutdown() {
           if (clusterWatcher != null && clusterWatcher.edsBalancer != null) {
             clusterWatcher.edsBalancer.shutdown();
+            xdsClient.cancelClusterDataWatch(cdsConfig.name, clusterWatcher);
```

Comment:
So this one doesn't have to be dependent on `clusterWatcher.edsBalancer != null` right? Or am I missing something?
```
          if (clusterWatcher != null) {
            if (clusterWatcher.edsBalancer != null) {
              clusterWatcher.edsBalancer.shutdown();
            }
            xdsClient.cancelClusterDataWatch(cdsConfig.name, clusterWatcher);
          }
```

Reply:
Thanks for the catch. Fixed.